docs_user/assemblies/assembly_prepare_director_operator_and_rhoso_for_adoption_process.adoc

```diff
@@ -0,0 +1,424 @@
+----
+$ CONTROLLER_NODE=$(oc -n $OSPDO_NAMESPACE get vmi -ojson | jq -r '.items[0].status.nodeName')
+export CONTROLLER_NODE
+----
+
+
+. If you are using a combination master/worker cluster with a high availability OSPdO environment,
+configure OSPdO to use a single {OpenStackShort} Controller node:
+
+.. Disable stonith:
++
+----
+$ stonith_resources=$(sudo pcs stonith status|grep Started|awk '{print $2}')
+for in in stonith_resources ;do CONTROLLER1_SSH sudo pcs stonith disable ;done
+----
+.. Put non-Controller nodes in maintenance:
++
+----
+$CONTROLLER1_SSH sudo pcs node maintenance <controller-1> <controller-2>
+----
++
+* Replace `<controller-1>` and `<controller-2>` with the names of the Controller nodes in your environment.
+
+.. Update the quorum for single node:
++
+----
+$CONTROLLER1_SSH sudo systemctl stop corosync
+$CONTROLLER2_SSH sudo systemctl stop corosync
+$CONTROLLER3_SSH sudo systemctl stop corosync
+$CONTROLLER1_SSH sudo pcs quorum update last_man_standing=1 wait_for_all=1
+$CONTROLLER1_SSH sudo systemctl restart corosync
+----
+.. Restart the Pacemaker cluster:
++
+----
+$CONTROLLER1_SSH sudo pcs cluster stop
+$CONTROLLER1_SSH sudo pcs cluster start
+----
+.. Remove the OSPdO node network configuration policies from the other two nodes that are not running the Controller VM by placing a `nodeSelector` on the {OpenShiftShort} node that contains the Controller VM:
++
+----
+$ for i in br-ctlplane br-ex br-osp; do oc patch osnetconfig openstacknetconfig --type json -p '[{"op": "replace", "path": "/spec/attachConfigurations/'$i'/nodeNetworkConfigurationPolicy/nodeSelector", "value": {"kubernetes.io/hostname": "'$CONTROLLER_NODE'"}}]'; done
+----
+
+== Preparing RHOSO for adoption
+
+When deploying a RHOSO env. usign the same master nodes, make sure
+to take the following steps:
+
+Get the names of the other two nodes that we will use for NG
+[,bash]
+----
+NG_NODES=$(oc get nodes -o name | grep -v $CONTROLLER_NODE | sed 's#node/##g' | tr '\n' ' ')
+----
+Extract nodes to be used for OSP 18 installation and label nodes.
+[,bash]
+----
+OSP18_NODE1=$(echo "${NG_NODES}" | cut -d ' ' -f 1)
+export OSP18_NODE1
+OSP18_NODE2=$(echo "${NG_NODES}" | cut -d ' ' -f 2)
+export OSP18_NODE2
+oc label nodes "${OSP18_NODE1}" type=openstack
+oc label nodes "${OSP18_NODE2}" type=openstack
+----
+
+Create a new namespace to use for installation of RHOSO.  RHOSO must be installed in a different namespace from OSPdO.
+[,bash]
+----
+oc get namespace ${OSP18_NAMESPACE} 2>/dev/null || {
+    oc create namespace ${OSP18_NAMESPACE} || {
+        echo "Failed to create namespace ${OSP18_NAMESPACE}"
+        exit 1
+    }
+}
+----
+
+Create custom NNCPs for NG.  The NNCPs will use the nodeSelector attribute to be constrained to OSP18_NODE1/2.  The new NNCP will mirror the existing NNCPs.  For this document, the subnets ranges are being reused.  In addition, OSPdO’s use of bridges for the control and external interfaces is duplicated.
+
+[,bash]
+----
```

Comment:
```suggestion
. Create custom node network configuration policies for NG: 
+
[NOTE]
The node network configuration policies use the `nodeSelector` attribute to be constrained to `OSP18_NODE1/2`.  The new node network configuration policy mirrors the existing node network configuration policies.  In the following example, the subnets ranges are reused.  In addition, OSPdO’s use of bridges for the control and external interfaces is duplicated.
+
----
```
@pinikomarov What is OSP18_NODE1/2`?
And what do we mean by "OSPdO’s use of bridges for the control and external interfaces is duplicated"? Does that mean that the same bridge is used for the control interface AND the external interface?

Reply:
During adoption the source and dest env must have network connectivity on the different OSP vlans like control-plane,
so here we're creating a new bridge , this time on the rhoso namespace (so it's seperate from the ospdo one), it will be on the same nic but have a subnet range which is a continuation of the ospdo one.
i.e. 
ospdo: br-ctlplane bridge on nic enp1s0, with subnet range 172.22.0.150-250 
<=> 
rhoso: ospdo br-ctlplane bridge on nic enp1s0, with subnet range 172.22.0.100-150
ref. https://github.com/atyronesmith/ospdo-migration/blob/3514c483eca9164fc49494ff522c8efb88f175ca/networks.txt